### PR TITLE
SDA .nc file output update

### PR DESCRIPTION
**Pull request by kzarada**
## Description
For not removing the `nc` output files of model simulations in SDA, @para2x was renaming the outputs to their name + a timestamp. But there was a bug in the code that this PR is fixing.

## Review Time Estimate

- [ ] Immediately
- [ ] Within one week
- [ X] When possible
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue) 
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the CHANGELOG.md.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Comment by para2x** [comment]
This looks good to me and it's what we discussed. Did you also get to test it ?

**Comment by kzarada** [comment]
@para2x yes, I ran it and it worked properly

File: modules/assim.sequential/R/sda.enkf_refactored.R [hunk 1]
```diff
@@ -350,19 +350,24 @@ sda.enkf <- function(settings,
...
+      full.names = TRUE)
+   files <-  files[grep(pattern = "SDA*", files, invert = TRUE)]
+    
+    
+    purrr::walk(files,
       function(.x){
         file.rename(.x ,
                     file.path(dirname(.x),
-                              paste0(gsub(" ", "", as.character(names(obs.mean)[t])),
+                              paste0("SDA_",gsub(" ", "", as.character(names(obs.mean)[t])),
```
Comment (infotroph) [inline modules/assim.sequential/R/sda.enkf_refactored.R]: Are all files in `files` guaranteed to live in different directories? If any of them share a `dirname`, they'll be given the same name and therefore overwrite each other. Or am I missing something?

Also, less importantly: `file.rename` accepts vectors, and `names()` is documented to return characters if it returns anything, so I *think* you can drop both `walk` and `as.character` and simplify this to

```
file.rename(
  files,
  file.path(
    dirname(files),
    paste0("SDA_", gsub(" ", "", names(obs.mean)[t]), ".nc"))
```

Comment (para2x) [reply modules/assim.sequential/R/sda.enkf_refactored.R]: I haven't seen any of our models to produce more than one `nc` output file per dir, but that's a good point and we can slightly modify that to use the `basename` but then we need the `walk` again something like this:
`paste0("SDA_", basename(.x), "_", names(obs.mean)[t], ".nc")`
>

Comment (para2x) [reply modules/assim.sequential/R/sda.enkf_refactored.R]: oh seems like the `basename` function is also vectorized and yeah seems like we don't the `walk` at all.

Comment (para2x) [reply modules/assim.sequential/R/sda.enkf_refactored.R]: @kzarada could you please give a try to this replacing the whole `walk` see if it works as expected :
`
file.rename(
    files,
    file.path(
       dirname(files),
       paste0("SDA_", basename(files), "_", gsub(" ", "", names(obs.mean)[t]), ".nc")
     )
 )
`

Comment (kzarada) [reply modules/assim.sequential/R/sda.enkf_refactored.R]: That code worked- the output files look like this 'SDA_2019.nc_2019-11-0500:00:00.nc'

Comment (kzarada) [reply modules/assim.sequential/R/sda.enkf_refactored.R]: I pushed the update

**Comment by kzarada** [comment]
@infotroph those were accidentally added (I forgot to create a new branch with this update since it was still WCr SDA related)- not relevant to this bug fix but they are part of the WCr SDA. I just added a shell file to run the forecast automatically, updated the workflow to work with the automation, and increased the ensemble size

**Comment by mdietze** [comment]
@kzarada those accidentally added files all sound like things we want to pull in, so lets go ahead and keep them in this PR.

**Pull request merged**